C/common/CMakeLists.txt

```diff
@@ -20,15 +20,44 @@ include_directories(SYSTEM ${Boost_INCLUDE_DIR})
 # Find source files
 file(GLOB SOURCES *.cpp)
 
+# Find Python.h 3.5 header file
+set(_PYTHON_3.5_INCLUDES /usr/include/python3.5m /usr/include/python3.5m)
+list(APPEND _PYTHON_3.5_INCLUDES /usr/include/python3.5 /usr/include/python3.5)
+find_path(Python3.5_INCLUDE NAMES Python.h PATHS ${_PYTHON_3.5_INCLUDES})
+if (NOT Python3.5_INCLUDE)
+        message(WARNING
+        " Python 3.5 header file 'Python.h' not found in /usr/include. \n"
+        " Target '${PROJECT_NAME}' filter is not built.")
+        return()
+else()
+        message(STATUS "Found Python 3.5 header file 'Python.h' in " ${Python3.5_INCLUDE})
+endif()
+
+# Find Python 3.5 library
+find_library(FoundPython_3.5 NAMES python3.5m python35m python3.5 python35)
+if (NOT FoundPython_3.5)
+        message(FATAL_ERROR "Python 3.5 library not found.")
+        return()
+else()
+        message(STATUS "Found Python 3.5 library in " ${FoundPython_3.5})
+endif()
+
 # Include header files
 include_directories(include ../services/common/include ../common/include ../thirdparty/rapidjson/include ../thirdparty/Simple-Web-Server)
 
+# Add Python 3.5 header files
+include_directories(${Python3.5_INCLUDE})
+
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/../lib)
 
 # Create shared library
 add_library(${PROJECT_NAME} SHARED ${SOURCES})
 target_link_libraries(${PROJECT_NAME} ${UUIDLIB})
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
+
+# Add Python 3.5 library
+target_link_libraries(${PROJECT_NAME} -lpython3.5m)
```

Comment:
On an i5 laptop with 16 GB RAM:
Runtime memory requirement (Pss) for benchmark south service has increased by 360 KB because of linking libpython3.5m.so for a total Pss of around 1440 KB.

It seems it is not straightforward to load libpython3.5m.so in a deferred manner.

I found following via a google search:

Runtime linker does support lazy binding of symbols, but not lazy loading of libraries. What this means is that each of the shared objects which an executable requires are loaded when the program starts, but the symbols within the program are not resolved to the loaded libraries until they are first referenced.

The only way to have a library loaded after program startup, for example in response to some command line option, configuration or other dynamic condition, is to call dlopen and then dlsym for each used symbol from that library.

And dlopen/dlsym way might not be very straightforward since we are using lots of symbols from that library.

Reply:
Ok sure, I'll look into this.

Reply:
This is done. libpython3.5m is presently being linked to the libsouth-plugin-python-interface.so library dynamically.